TILE_AUTHOR_GUIDE.md

```diff
@@ -0,0 +1,598 @@
+- `job` reads, interpolates, and formats a product template part from `./jobs`
+- `runtime_config` reads, interpolates, and formats a product template part from `./runtime_configs`
+
+Other functions:
+- `tile` _TODO function documentation_
+- `select`  _TODO function documentation_
+- `regexReplaceAll`  _TODO function documentation_
+
+See the [crhntr/hello-tile/base.yml](https://github.com/pivotal/hello-tile/blob/c6b59dcb1118c9b2f5d4fbf836842ce4033baa80/base.yml#L29C1-L30C1) for some use of the above functions.
+The property definition in [crhntr/hello-tile/properties/hello.yml](https://github.com/pivotal/hello-tile/blob/c6b59dcb1118c9b2f5d4fbf836842ce4033baa80/properties/hello.yml#L2-L5)
+in referenced in `base.yml` using `$( property "port" )`.
+Most other product template part functions behave similarly.
+
+## <a id="bosh-release-tarballs"></a> Managing BOSH Release Tarballs
+
+`kiln fetch` downloads BOSH Release Tarballs from any of the following "sources"
+and puts them in a "./releases" directory.
+
+Before Kiln can help you manage the BOSH Releases you put in the tile, you need to upload BOSH Release tarballs to a place accessable to Kiln.
+
+See the sources below to decide which is right for your release.
+
+Unlike Tile Generator.
+Kiln does not create releases.
+The Kiln way is to 
+- have BOSH releases each have their own source control
+- have CI build and upload (finalized) BOSH release tarballs somewhere
+- have a tile source repository
+  - in the tile source repository specify how Kiln can get the BOSH release tarballs
+
+While the following examples start from empty directories and mostly use S3 and BOSH.io.
+There are similar simple scripts for a small test tile illustrating similar usage patterns to the follwoing example.
+See [hello-tile](https://github.com/crhntr/hello-tile).
+Hello Tile consumes a single custom BOSH Release, [hello-release](https://github.com/crhntr/hello-release), from a GitHub release.
+It does not upload the release to PivNet but adds the built tile to a GitHub Release.
+
+#### <a id="kiln-fetch-example"></a> **EXAMPLE** Using Kiln to Download BOSH Release Tarballs
+This starts from an empty directory and downloads the latest BPM release from bosh.io.
+Note, the Kilnfile and Kilnfile.lock (unfortunately/frustratingly) must be created manually.
+
+```sh
+# Create and go into an empty directory
+mkdir -p /tmp/try-kiln-fetch
+cd !$
+mkdir -p releases
+touch release/.gitkeep # not required but a good idea
+
+# Hack a Kilnfile and Kilnfile lock
+echo '{"release_sources": [{type: bosh.io}], "releases": [{"name": "bpm"}]}' > Kilnfile
+yq '{"releases": [. | {"name": "bpm", "version": .version, "sha1": .sha, "remote_source": "bosh.io", "remote_path": .remote_path}]}' <(kiln find-release-version --release=bpm) > Kilnfile.lock
+
+# Call Kiln fetch
+kiln fetch
+
+# See the fetched release
+stat releases/bpm*.tgz
+```
+
+The files should look something like these
```

Comment:
I see a JSON instead of yaml on this step
![image](https://github.com/pivotal-cf/kiln/assets/25727596/a1512a84-af5b-4771-9a75-630fe6cc2d4b)

Reply:
I decided to write JSON files because I could do it one line and YAML is mostly a super set of JSON.

Reply:
From the PoV of a consumer of the doc the inconsistency between the result of the commands and the guide is what I was mentioning, you talk about `look s something like these` but we get something different.

I understand that both work just wanted to point out that what's written in the document does not reflect what is happening in my machine.

Reply:
Okay. That makes sense. I'll pipe the json through yq to make it more consistent.

Reply:
The generated Kilnfile and Kilnfile.lock now match the expected output. I will resolve the conversation.